### ISSUE TITLE
render wrapper error TS2769: No overload matches this call

### ISSUE THREAD

**Issue by akornatskyy** [testing-library/react-testing-library#972]
- `@testing-library/react` version: 12.1.1
- Testing Framework and version: jest@27.2.4
  
- typescript@4.4.3

### Relevant code or config:

```js
import {MemoryRouter as Router} from 'react-router-dom';
// ...
render(<Whatever />, {wrapper: Router});
```

### What you did:

Use `Router` as `wrapper`.

### What happened:

typescript compile error.

### Problem description:

```typescript
 Overload 1 of 2, '(ui: ReactElement<any, string | JSXElementConstructor<any>>, options: RenderOptions<typeof import(".../node_modules/@testing-library/dom/types/queries"), HTMLElement>): RenderResult<...>', gave the following error.
Type 'typeof MemoryRouter' is not assignable to type 'ComponentType<{ children: ReactElement<any, string | JSXElementConstructor<any>>; }> | undefined'.
Type 'typeof MemoryRouter' is not assignable to type 'ComponentClass<{ children: ReactElement<any, string | JSXElementConstructor<any>>; }, any>'.
Types of parameters 'props' and 'props' are incompatible.
Type '{ children: ReactElement<any, string | JSXElementConstructor<any>>; }' is not assignable to type 'MemoryRouterProps | Readonly<MemoryRouterProps>'.
Type '{ children: ReactElement<any, string | JSXElementConstructor<any>>; }' has no properties in common with type 'Readonly<MemoryRouterProps>'.
Overload 2 of 2, '(ui: ReactElement<any, string | JSXElementConstructor<any>>, options?: Omit<RenderOptions<typeof import(".../node_modules/@testing-library/dom/types/queries"), HTMLElement>, "queries"> | undefined): RenderResult<...>', gave the following error.
Type 'typeof MemoryRouter' is not assignable to type 'ComponentType<{ children: ReactElement<any, string | JSXElementConstructor<any>>; }> | undefined'.
Type 'typeof MemoryRouter' is not assignable to type 'ComponentClass<{ children: ReactElement<any, string | JSXElementConstructor<any>>; }, any>'.
```

**Comment by akornatskyy** [comment]
That is still the issue with **12.1.2**.

**Comment by akornatskyy** [comment]
@eps1lon : just in case you have missed the last comment above, this error still can be reproduced with `12.1.2` (I cannot reopen this issue). advise, please.

**Comment by tb-viktor** [comment]
Have you managed to make some progress on this issue?

We are still getting it after upgrading. These are the relevant dependencies:

```
"@testing-library/jest-dom": "^5.16.1",
"@testing-library/react": "^12.1.2",
"@types/react": "^17.0.38",
"@types/react-dom": "^17.0.11",
"@types/react-router-dom": "^5.3.2",
"react": "^17.0.2",
"react-dom": "^17.0.2",
"react-router-dom": "^5.2.0",
"react-scripts": "^5.0.0",
"typescript": "4.3",
```

As in the description above, we've declared the `MemoryRouter` as the wrapper of our render:

```
render(<Component />, { wrapper: MemoryRouter });
```

Which makes TS throw the issue:

```
No overload matches this call.
  Overload 1 of 2, '(ui: ReactElement<any, string | JSXElementConstructor<any>>, options: RenderOptions<typeof import("C:/Users/Tomislav/viktor-frontend/node_modules/@testing-library/dom/types/queries"), HTMLElement>): RenderResult<...>', gave the following error.
    Type 'typeof MemoryRouter' is not assignable to type 'JSXElementConstructor<{ children: ReactElement<any, string | JSXElementConstructor<any>>; }> | undefined'.
      Type 'typeof MemoryRouter' is not assignable to type 'new (props: { children: ReactElement<any, string | JSXElementConstructor<any>>; }) => Component<any, any, any>'.
        Types of parameters 'props' and 'props' are incompatible.
          Type '{ children: ReactElement<any, string | JSXElementConstructor<any>>; }' is not assignable to type 'MemoryRouterProps | Readonly<MemoryRouterProps>'.
            Type '{ children: ReactElement<any, string | JSXElementConstructor<any>>; }' has no properties in common with type 'Readonly<MemoryRouterProps>'.
  Overload 2 of 2, '(ui: ReactElement<any, string | JSXElementConstructor<any>>, options?: Omit<RenderOptions<typeof import("C:/Users/Tomislav/viktor-frontend/node_modules/@testing-library/dom/types/queries"), HTMLElement>, "queries"> | undefined): RenderResult<...>', gave the following error.
    Type 'typeof MemoryRouter' is not assignable to type 'JSXElementConstructor<{ children: ReactElement<any, string | JSXElementConstructor<any>>; }> | undefined'.
      Type 'typeof MemoryRouter' is not assignable to type 'new (props: { children: ReactElement<any, string | JSXElementConstructor<any>>; }) => Component<any, any, any>'.ts(2769)
```

**Comment by eps1lon** [comment]
This issue was fixed as of `@types/react@17.0.38` and `@testing-library/react@12.1.2`:

```tsx
import * as React from 'react';
import { render, screen } from '@testing-library/react';
import {MemoryRouter as Router} from 'react-router-dom';

function Whatever() {
    return null;
}

render(<Whatever />, {wrapper: Router});
```

[Playground Link](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAKjgQwM5wEoFNkGN4BmUEIcA5FDvmQNwBQokscA3nJQHYAmWUANHFS5KWDnAC+cIiXIABGFlQxgHAOYBaADbAARlGRQAngHpKeGLQbho8FgFksIaIYwQArgqgp0rj70nSpBRUMOrEflDqXCSWdARuHPjAEGIA6gAWyAoAbrwAFACUrHRwpexYMG5QYhxumpr04nR0nDxQeQA8GVlYuV7GAHwCLADu+mBgvABcmO6e4gU0QA)

If the issue persists please include a cloneable repository reproducing this issue. You may have to upgrade `react-router-dom` to the latest version.

**Comment by its-vincesanity** [comment]
In 18.2.0, I have the same Issue...
